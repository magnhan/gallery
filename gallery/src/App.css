.grid-container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-row: repeate(5, 1fr);
  grid-gap: 5px;
  min-height: 100vh; /*stretches to fill the screen*/
  background-color: beige;
  border-radius: 15px;
}

.grid-container > div {
  text-align: center;
  padding: 20px;
  font-size: 30px;
  justify-self: stretch;
}
.Header {
  grid-column: 1/6;
}
.Menu {
  grid-column: 1/2;
  grid-row: 2 / 80;
}
.DisplayArt {
  grid-column: 2/6;
  grid-row: 2 / 80;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
}
.Footer {
  grid-column: 1/6;
}

.SVG {
  background-color: #444;
  font-size: 150%;
  height: 300px;
  width: 280px;
  justify-self: center;
}

.trumpet{
  fill: #e1ab36;
  fill-rule: evenodd;
}

.SVG1 {
  grid-column: 1;
  grid-row: 1;
}
.SVG2 {
  grid-column: 2;
  grid-row: 1;
}
.SVG3 {
  grid-column: 3;
  grid-row: 1;
}
.SVG4 {
  grid-column: 1;
  grid-row: 2;
}
.SVG5 {
  grid-column: 2;
  grid-row: 2;
}
.SVG6 {
  grid-column: 3;
  grid-row: 2;
}

.SVG_picture {
  height: 250px;
  width: 280px;
  background-color: white;
}

.text {
  color: #fff;
  font-size: x-small;
}

@media only screen and (max-width: 650px) {
  .Menu {
    grid-column: 1/6;
    grid-row: 2 / 30;
  }
  .DisplayArt {
    grid-column: 1/6;
    grid-row: 30 / 100;
  }
}

/* On screens that are 350px or less */
@media only screen and (max-width: 350px) {
  .Menu {
    grid-column: 1/6;
    grid-row: 2 / 30;
  }
  .DisplayArt {
    grid-column: 1/6;
    grid-row: 30 / 100;
  }
  .SVG {
    height: 200px;
    width: 200px;
    justify-items: stretch;
  }
  .SVG_picture {
    height: 160px;
    width: 200px;
    justify-items: center;
  }
}
