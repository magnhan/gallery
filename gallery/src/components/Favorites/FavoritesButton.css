#favorites {
  fill: #d64550;
  transform: scale(0.7);
}

.SaveButton{
  fill: white;
  border: 20p;
  transform: scale(0.7);
}

.SaveButton:hover{
  fill: #d64550;
}

#favBtn{
  background: none;
  border: none;
  cursor: pointer;
  padding: 0px;
  appearance: none;
}
