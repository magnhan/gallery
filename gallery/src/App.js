import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(){
    super();
      
    this.state = {
      string: 'Hello this is our gallery'
    }
  }
  render(){
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            {this.state.string}
          </p>
          <button onClick={()=> this.setState({string: 'Text is changed'})}>Change text</button>
        </header>
      </div>
    );
  }
  
}

export default App;
