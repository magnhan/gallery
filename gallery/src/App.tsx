import React, { Component } from "react";
import "./App.css";
import "./index.css";
import Header from "./components/Header";
import Menu from "./components/Menu";
import DisplayArt from "./components/DisplayArt";
import Footer from "./components/Footer";

export default function App() {
  return (
    <div className="grid-container">
      <Header />
      <Menu />
      <DisplayArt />
      <Footer />|
    </div>
  );
}
