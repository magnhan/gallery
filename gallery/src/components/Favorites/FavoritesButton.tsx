import React from "react";
import { NavLink } from "react-router-dom";
import "./FavoritesButton.css";

export default function FavoritesButton() {
  return (
    <div>
      <NavLink to="/Favorites">
        <div className="click">
          {" "}
          <span className="fa fa-star-o"></span>
        </div>
      </NavLink>
    </div>
  );
}
