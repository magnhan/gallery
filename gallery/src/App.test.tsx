import React from "react";
import App from "./App";
import FavoritesButton from "./components/Favorites/FavoritesButton";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

//snapshot test for favorite button
test("First snapshottest", () => {
  const component = renderer.create(<FavoritesButton />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//smoke test for components
test("Renders without crashing", () => {
  const smokeTest = document.createElement("smokeTest");
  ReactDOM.render(<App />, smokeTest);
});
