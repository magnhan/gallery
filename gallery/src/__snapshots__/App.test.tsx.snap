// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshottest 1`] = `
<div
  className="FavoritesButton"
>
  <svg
    height="70px"
    id="favorites"
    width="70px"
  >
    <path
      d="M 46.296296,51.906272 L 31.916351,42.474649 L 17.502712,51.8547 L
 22.029072,35.264028 L 8.654054,24.454438 L 25.831443,23.632463 L
 31.978866,7.5717174 L 38.068716,23.65438 L 55.243051,24.537884 L
 41.829396,35.299492 L 46.296296,51.906272 z "
    />
  </svg>
</div>
`;
